x-pack/plugins/ml/public/application/jobs/jobs_list/components/edit_job_flyout/tabs/job_details.js

```diff
@@ -172,6 +186,20 @@ export class JobDetails extends Component {
               disabled={datafeedRunning}
             />
           </EuiFormRow>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.jobsList.editJobFlyout.jobDetails.modelSnapshotRetentionDaysLabel"
+                defaultMessage="Model snapshot retention days"
+              />
+            }
+          >
+            <EuiFieldNumber
```

Comment:
I found that you can still save values with fractional numbers of days, but the back-end then floors the input, so I think the behaviour is ok. This happens even if you add a `step={1}` on the `EuiFieldNumber`. The input does indicate invalid input although the Save button remains enabled:

![image](https://user-images.githubusercontent.com/7405507/84509593-d889e900-acbb-11ea-8069-497fb7abebf5.png)

Reply:
I've wrapped the `onChange` input in a `Math.floor` so when the user inputs a float, it immediately changes to an int.